src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
+#if 0
+u8 D_801BFDE8[PLAYER_MASK_MAX - 1] = {
+    ITEM_MASK_TRUTH,         // PLAYER_MASK_TRUTH
+    ITEM_MASK_KAFEIS_MASK,   // PLAYER_MASK_KAFEIS_MASK
+    ITEM_MASK_ALL_NIGHT,     // PLAYER_MASK_ALL_NIGHT
+    ITEM_MASK_BUNNY,         // PLAYER_MASK_BUNNY
+    ITEM_MASK_KEATON,        // PLAYER_MASK_KEATON
+    ITEM_MASK_GARO,          // PLAYER_MASK_GARO
+    ITEM_MASK_ROMANI,        // PLAYER_MASK_ROMANI
+    ITEM_MASK_CIRCUS_LEADER, // PLAYER_MASK_CIRCUS_LEADER
+    ITEM_MASK_POSTMAN,       // PLAYER_MASK_POSTMAN
+    ITEM_MASK_COUPLE,        // PLAYER_MASK_COUPLE
+    ITEM_MASK_GREAT_FAIRY,   // PLAYER_MASK_GREAT_FAIRY
+    ITEM_MASK_GIBDO,         // PLAYER_MASK_GIBDO
+    ITEM_MASK_DON_GERO,      // PLAYER_MASK_DON_GERO
+    ITEM_MASK_KAMARO,        // PLAYER_MASK_KAMARO
+    ITEM_MASK_CAPTAIN,       // PLAYER_MASK_CAPTAIN
+    ITEM_MASK_STONE,         // PLAYER_MASK_STONE
+    ITEM_MASK_BREMEN,        // PLAYER_MASK_BREMEN
+    ITEM_MASK_BLAST,         // PLAYER_MASK_BLAST
+    ITEM_MASK_SCENTS,        // PLAYER_MASK_SCENTS
+    ITEM_MASK_GIANT,         // PLAYER_MASK_GIANT
+    ITEM_MASK_FIERCE_DEITY,  // PLAYER_MASK_FIERCE_DEITY
+    ITEM_MASK_GORON,         // PLAYER_MASK_GORON
+    ITEM_MASK_ZORA,          // PLAYER_MASK_ZORA
+    ITEM_MASK_DEKU,          // PLAYER_MASK_DEKU
+};
+#endif
+
+u8 func_80122ED8(s32 maskIdMinusOne) {
+    return D_801BFDE8[maskIdMinusOne];
+}
+
+u8 Player_CurMaskToItemId(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (player->currentMask != PLAYER_MASK_NONE) {
+        return func_80122ED8(player->currentMask - 1);
+    }
+
+    return ITEM_NONE;
+}
+
+void func_80122F28(Player* player) {
+    if ((player->actor.category == ACTORCAT_PLAYER) &&
+        (!(player->stateFlags1 & (PLAYER_STATE1_20000000 | PLAYER_STATE1_800000 | PLAYER_STATE1_200000 |
+                                  PLAYER_STATE1_800 | PLAYER_STATE1_400))) &&
+        (!(player->stateFlags2 & PLAYER_STATE2_1))) {
+        if (player->doorType < 0) {
+            ActorCutscene_SetIntentToPlay(0x7C);
+        } else {
+            ActorCutscene_SetIntentToPlay(0x7D);
+        }
+    }
+}
+
+s32 func_80122F9C(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    return (player->stateFlags2 & PLAYER_STATE2_80000) && player->unk_AE7 == 2;
+}
+
+s32 func_80122FCC(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    return (player->stateFlags2 & PLAYER_STATE2_80000) && (player->unk_AE7 == 1 || player->unk_AE7 == 3);
+}
+
+void func_8012300C(GlobalContext* globalCtx, s32 arg1) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    player->unk_B2B = arg1;
+}
+
+void func_8012301C(Player* player, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    s32 pad;
+
+    player->unk_AE7++;
+
+    if (player->unk_AE7 == 2) {
+        s16 objectId = gLinkFormObjectIndexes[((void)0, gSaveContext.playerForm)];
```

Comment:
```suggestion
        s16 objectId = gPlayerFormObjectIndexes[((void)0, gSaveContext.playerForm)];
```
Unless there's good motivation to use `Link` here

Reply:
Ohh nice catch. That variable is already on master and I forgot to rename it